docs/concepts/artifact.mdx

```diff
@@ -0,0 +1,239 @@
+      "digest": "sha256:d2a84f4b8b650937ec8f73cd8be2c74add5a911ba64df27458ed8229da804a26",
+      "size": 12,
+      "annotations": {
+        "org.opencontainers.image.title": "hello.txt"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2023-08-03T00:21:51Z"
+  }
+}
+```
+
+### Artifacts with config
+
+Clients can use `config.mediaType` property  to
+declare the artifact type. The `config.mediaType` may be used when artifacts have a valid config blob and
+the mediaType matches the config blob and type of the artifact. For example the
+artifact type below for the helm chart is `application/vnd.cncf.helm.config.v1+json`
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cncf.helm.config.v1+json",
+    "digest": "sha256:34bf03806938a59ee7dc3e2c33e314d0eaef573313ff9dcc677113502d568523",
+    "size": 145
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.cncf.helm.chart.content.v1.tar+gzip",
+      "digest": "sha256:4d80464e9d8e9f3ba92e6ead6d3b5afd0532cb0a81e980599a0bced99fdc6e01",
+      "size": 3763
+    }
+  ]
+}
+```
+
+### Annotations only artifacts
+
+Artifacts may store metadata in the manifest as annotations and do not have a config or blobs.
+For these the `artifactType` property is used to declare the type of the artifact.
+The config property is required in the manifest for maximum compatiblity an empty layer is also created.
+as per the  [empty descriptors guidance][empty-descriptor].  
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.example+type",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.empty.v1+json",
+      "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+      "size": 2
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2023-01-02T03:04:05Z",
+    "com.example.info": "useful-info"
+  }
+}
+```
+
+### Harnessing Image Indexes
+
+One of the key advantages of OCI artifacts is their ability to utilize [Image Indexes][image-index],
+allowing the bundling of various OCI artifacts. An example use case if is when you want to represent
+multiple SBOMs of different architectures under one unified artifact.
+
+The Image Index is a higher-level construct in the OCI Image Specification that can point to
+multiple Image Manifests, each suitable for a different platform or architecture. It helps to
+group related artifacts together, providing a single entry-point for accessing any specific
+artifact depending on the required architecture.
+
+Below is an example of an OCI Image Index for an artifact that represents multiple SBOMs
+of different architectures:
+
+```json 
```

Comment:
This seems like a fairly controversial example. 
SBOMs for platform specific images would be more generically applicable if the SBOM for each image is referenced by a platform specific SBOM artifact, with a `subject` of the platform specific image. That's generically accessible, vs. this concept of using an index. 

I can't think of a better example, but would suggest not using this as it will be a template for others to use, which I think would be a mistake.

Reply:
I can go for a generic example if that helps. But I'm glad that this is stirring up the conversation since finding a real-world example for the this scenario is turning out to be hard. 
@toddysm what do you think or do you have a suggestion for good example?  

Again this is not the OCI specification and hence I feel we can use actual examples rather than just copy examples from the spec.

Reply:
One of the drawbacks of using an index based approach is race conditions (The index is a form of a materialised view that is calculated by the last thing that updates it and two things writing at the same time will ignore the other). Cosign suffers from this because of their .sig and .att tags being an "index" with layers per new item. So any index based example should be for a "write once"  scenario.

A good example would be something like this https://slsa.dev/verification_summary/v1

Imagine a bunch of people having to sign off on a container image being promoted from UAT to Production. It happens asynchronously and evidence is attached to the artefact.

Once everyone signs off, a workflow management software creates an index of all the evidence and creates a new artefact similar to the VSA above and attaches it.

For gatekeeping rules, the production environment only needs to check for the existence of that index and criteria in it. Not each individual approver's evidence.

Reply:
For now I've removed the index example. ORAS currently doesn't have way to create indexes and is a missing part of the picture.

Reply:
Removing for now is not a bad thing. The one that was initially interesting was CNAB, where you want to bundle a bunch of artifacts together. Howeer, they also needed cross repo support, which isn't currently supported in index, I believe. 
Either way, removing now, until we add support in ORAS is a good call.